middleman-core/lib/middleman-core/preview_server.rb

```diff
@@ -194,7 +194,9 @@ def setup_webrick(is_logging)
             http_opts[:SSLPrivateKey] = OpenSSL::PKey::RSA.new File.read ssl_private_key
           else
             # use a generated self-signed cert
-            http_opts[:SSLCertName] = [%w(CN localhost), %W(CN #{host})].uniq
+            cert, key = create_self_signed_cert(1024, [["CN", host]], "Middleman Preview Server")
```

Comment:
A bit picky, but - although it's local - 1024 is considered to be broken... We should a least use 2048.